### DISCUSSION TITLE
How to type annotate a binary operator?

### DISCUSSION THREAD

**Discussion by LeeeeT** [microsoft/pyright#8962] [Q&A]
I have a code that defines a binary operator like this:

```python
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class MyClass:
    x: int

    def __add__(self, other: MyClass, /) -> MyClass:
        if isinstance(other, MyClass):
            return MyClass(self.x + other.x)
        return NotImplemented
```

Due to how binary operators work in python, I am forced to wrap the actual implementation in `if isinstance` block and return `NotImplemented` if the check fails.

This worked without errors prior to version 1.1.380. On version, 1.1.380 I'm getting this error:

```
file.py:11:12 - error: Unnecessary isinstance call; "MyClass" is always an instance of "MyClass"
```

So, my question is: how to properly type annotate `__add__`, given the special behavior of binary operators in python?

**Comment by erictraut** [comment]
Your code looks fine to me. You would need to suppress the `reportUnnecessaryIsInstance` error in this case.
